lib/src/session.dart

```diff
@@ -184,3 +202,38 @@ class SessionProgressNotificationsController {
     _token = null;
   }
 }
+
+/// @nodoc
+class SessionConnectionStateController {
+  final Session _session;
+  late final StreamController<ConnectionStateChange> _streamController;
+  int? _token;
```

Comment:
My usual reservations.

Reply:
Yeah, I looked at 2.17 finalizers and they don't seem to work the way I was hoping they will. I think we can still make it work by using weak references but I feel it's a bit of a risky change that I won't have the time to properly test and think through.

Reply:
This is related to what @nielsenko is working on. Should we drop both tasks for until after the release. Or should we fix both with the same fix

Reply:
I think we should merge this one as is since it's not inherently wrong - if the user cancels the subscription, the handle will be cleaned up and similarly to the progress notifiers, these are not expensive subscriptions to keep around.

When @nielsenko has a fix for the progress notifier, he can apply the same fix for the connection state notifiers, but that's not critical for the release.

Reply:
its ok by me